synapse/rest/client/sync.py

```diff
@@ -544,8 +544,13 @@ def serialize(events: Iterable[EventBase]) -> Awaitable[List[JsonDict]]:
                     event.room_id,
                 )
 
-        serialized_state = await serialize(state_events)
-        serialized_timeline = await serialize(timeline_events)
+        # No need to bundle aggregations for state events.
+        serialized_state = await serialize(state_events, bundle_aggregations=False)
+        # Only bundle aggregations if the room is limited, as clients could be
+        # missing events.
```

Comment:
This comment is confusing. Why would bundling aggregations be a problem if clients were missing events?

[The MSC doesn't really explain why we only bundle aggregations if the timeline is limited, either.]

Reply:
I believe the idea is that if the sync is limited we include the bundled aggregations so that the clients are able to display the event (with all reactions, etc.) without having to fetch the rest of the events.

Reply:
Maybe the following is clearer:

> Only bundle aggregations if the room is limited, otherwise the client should already have all the information necessary to render events.

Does that make sense?

Reply:
> > Only bundle aggregations if the room is limited, otherwise the client should already have all the information necessary to render events.
>
> Does that make sense?

Right yes, that makes more sense. Alternatively:

```suggestion
        # Don't bother to bundle aggregations if the timeline is unlimited, as
        # clients will have all the necessary information.
```